shared-routes2/app1/webpack.config.js

```diff
@@ -31,7 +33,35 @@ module.exports = {
       name: 'app1',
       filename: 'remoteEntry.js',
       remotes: {
-        app2: 'app2@http://localhost:3002/remoteEntry.js',
+        app2: `promise new Promise((resolve, reject) => {
```

Comment:
Use this instead.

Add this plugin 

```js
class AddRuntimeRequiremetToPromiseExternal {
  apply(compiler) {
    compiler.hooks.compilation.tap(
      "AddRuntimeRequiremetToPromiseExternal",
      (compilation) => {
        const RuntimeGlobals = compiler.webpack.RuntimeGlobals;
        if (compilation.outputOptions.trustedTypes) {
          compilation.hooks.additionalModuleRuntimeRequirements.tap(
            "AddRuntimeRequiremetToPromiseExternal",
            (module, set, context) => {
              if (module.externalType === "promise") {
                set.add(RuntimeGlobals.loadScript);
                set.add(RuntimeGlobals.require);
              }
            }
          );
        }
      }
    );
  }
}
```

then use 

```js
        promsie new Promise(function (resolve, reject) {
          var __webpack_error__ = new Error()
          if (typeof window[name] !== 'undefined') return resolve();
          __webpack_require__.l(
            "http://localhost:3002/remoteEntry.js"
            function (event) {
              if (typeof app2 !== 'undefined') return resolve();
              var errorType = event && (event.type === 'load' ? 'missing' : event.type);
              var realSrc = event && event.target && event.target.src;
              __webpack_error__.message =
                'Loading script failed.\n(' + errorType + ': ' + realSrc + ')';
              __webpack_error__.name = 'ScriptExternalLoadError';
              __webpack_error__.type = errorType;
              __webpack_error__.request = realSrc;
              reject(__webpack_error__);
            },
            app2,
          );
        })
```

Then handle the failover for the remotes, but use _webpack_require_.l for script injection

Reply:
```js
            shop: `promsie new Promise(function (resolve, reject) {
          var __webpack_error__ = new Error()
          if (typeof shop !== 'undefined') return resolve();
          __webpack_require__.l(
            "http://localhost:3001/_next/static/chunks/remoteEntry.js"
            function (event) {
              if (typeof shop !== 'undefined') return resolve();
              var errorType = event && (event.type === 'load' ? 'missing' : event.type);
              var realSrc = event && event.target && event.target.src;
              __webpack_error__.message =
                'Loading script failed.\\n(' + errorType + ': ' + realSrc + ')';
              __webpack_error__.name = 'ScriptExternalLoadError';
              __webpack_error__.type = errorType;
              __webpack_error__.request = realSrc;
              reject(__webpack_error__);
            },
            "shop",
          );
        }).then(function(){
         var proxy = {
         get: shop.get,
         init: shop.init
         }
         return proxy
        })`,

```

Reply:
@ScriptedAlchemy updated with your suggestion. I found easier alternatives but would be nice to document why the above approach is required / recommended